File: frontend/src/types/watchlist.ts
// 自选股分组
export interface WatchlistGroup {
  id: string;
  name: string;
  color: string;
  createdAt: number;
}

// 自选股票项
export interface WatchlistItem {
  symbol: string;
  name: string;
  addedAt: number;
  note?: string;
  groupIds: string[]; // 所属分组ID列表，支持多分组
}

// 完整的自选数据
export interface WatchlistData {
  groups: WatchlistGroup[];
  stocks: WatchlistItem[];
  version: number;
}

// 添加到自选时的参数
export interface AddToWatchlistParams {
  symbol: string;
  name: string;
  groupIds: string[];
  note?: string;
}

// 创建分组参数
export interface CreateGroupParams {
  name: string;
  color?: string;
}

// 预设分组颜色
export const GROUP_COLORS = [
  '#1890ff', // 蓝色
  '#52c41a', // 绿色
  '#faad14', // 黄色
  '#f5222d', // 红色
  '#722ed1', // 紫色
  '#13c2c2', // 青色
  '#eb2f96', // 粉色
  '#fa541c', // 橙色
];

// localStorage key
export const WATCHLIST_STORAGE_KEY = 'ai-quant-watchlists-v1';

File: app/models/watchlist.py
"""
自选数据模型
"""
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, Field


class WatchlistGroup(BaseModel):
    """自选分组"""
    id: str = Field(..., description="分组ID")
    name: str = Field(..., description="分组名称")
    color: str = Field(default="#1890ff", description="分组颜色")
    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: Optional[datetime] = Field(None, description="更新时间")


class WatchlistItem(BaseModel):
    """自选股票"""
    symbol: str = Field(..., description="股票代码")
    name: str = Field(default="", description="股票名称")
    group_ids: List[str] = Field(default_factory=list, description="所属分组ID列表")
    note: str = Field(default="", description="备注")
    created_at: datetime = Field(default_factory=datetime.now, description="添加时间")
    updated_at: Optional[datetime] = Field(None, description="更新时间")


class WatchlistData(BaseModel):
    """自选数据"""
    groups: List[WatchlistGroup] = Field(default_factory=list, description="分组列表")
    stocks: List[WatchlistItem] = Field(default_factory=list, description="股票列表")


class CreateGroupRequest(BaseModel):
    """创建分组请求"""
    name: str = Field(..., description="分组名称")
    color: str = Field(default="#1890ff", description="分组颜色")


class AddStockRequest(BaseModel):
    """添加股票请求"""
    symbol: str = Field(..., description="股票代码")
    name: str = Field(default="", description="股票名称")
    group_ids: List[str] = Field(default_factory=list, description="分组ID列表")
    note: str = Field(default="", description="备注")


class UpdateStockRequest(BaseModel):
    """更新股票请求"""
    group_ids: Optional[List[str]] = Field(None, description="分组ID列表")
    note: Optional[str] = Field(None, description="备注")


class RenameGroupRequest(BaseModel):
    """重命名分组请求"""
    name: str = Field(..., description="新名称")
